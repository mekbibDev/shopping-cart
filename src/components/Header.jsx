import { Link } from "react-router-dom"
export default function Header() {
  return (
    <header>
      <h1>SwiftCart Provisions Express</h1>
      <nav>
        <Link to="./Home">Home</Link>
        <Link to="./Shop">Shop</Link>
        <a href="#">Shopping Cart</a>
      </nav>
    </header>
  )
}