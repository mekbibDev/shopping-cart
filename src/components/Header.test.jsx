import { render } from "@testing-library/react";
import Header from "./Header";
import { BrowserRouter } from "react-router-dom";
import { describe, expect, it } from "vitest";

describe("Header", () => {
  it("matches the snapshot taken", () => {
    const { container } = render(
      <BrowserRouter>
        <Header> </Header>
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
