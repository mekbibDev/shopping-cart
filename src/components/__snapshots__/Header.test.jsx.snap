// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > matches the snapshot taken 1`] = `
<div>
  <header>
    <h1>
      SwiftCart Provisions Express
    </h1>
    <nav>
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/shop"
      >
        Shop
      </a>
      <div>
        <h1>
          cart:
          0
        </h1>
        <a
          href="/checkout"
        >
          Go to Checkout
        </a>
      </div>
    </nav>
  </header>
  <main
    style="align-self: center; max-width: 800px;"
  >
    <h2>
      Welcome to SwiftCart Provisions Express!
    </h2>
    <p>
      Discover a world of convenience at your fingertips with SwiftCart, your go-to destination for hassle-free shopping. Dive into a seamless online shopping experience where quality meets affordability. From the latest trends to everyday essentials, we've got you covered. Why SwiftCart?
    </p>
    <ol>
      <li>
        Unbeatable Selection: Explore a vast array of products from top brands across diverse categories. From fashion and electronics to home essentials, SwiftCart brings you a curated selection to cater to all your needs.
      </li>
      <li>
        Seamless Shopping Experience: Enjoy a user-friendly interface that makes your shopping journey smooth and enjoyable. Our intuitive design ensures you can find what you need effortlessly, saving you time and energy.
      </li>
      <li>
        Exclusive Deals and Discounts: At SwiftCart, we believe in offering our customers more for less. Take advantage of exclusive deals and discounts that make your shopping experience not only enjoyable but also budget-friendly.
      </li>
      <li>
        Swift Delivery: Need it fast? SwiftCart Provisions Express lives up to its name. Experience swift and reliable delivery services that bring your purchases right to your doorstep, ensuring you get what you need when you need it.
      </li>
      <li>
        Secure Payment Options: Your security is our priority. Shop with confidence, knowing that SwiftCart provides secure payment options to safeguard your transactions.
      </li>
    </ol>
    <p>
      Join the SwiftCart family today and redefine the way you shop. Convenience, quality, and affordability, that's the SwiftCart promise. Happy Shopping!
    </p>
  </main>
  <footer>
    <address>
      <h1>
        Contact us
      </h1>
      Have a question or need assistance? We're here to help! Feel free to reach out to us using the contact information below:
      <h2>
        Customer Support
      </h2>
      <nav>
        <a
          href="+1-800-123-4567"
        >
          1-800-123-4567
        </a>
        <a
          href="mailto:support@swiftcartexpress.com"
        >
          support@swiftcartexpress.com
        </a>
      </nav>
      <h2>
        Business enquires
      </h2>
      <nav>
        <a
          href="+1-800-765-4321"
        >
          1-800-765-4321
        </a>
        <a
          href="mailto:business@swiftcartexpress.com"
        >
          business@swiftcartexpress.com
        </a>
      </nav>
      <h2>
        Address
      </h2>
      SwiftCart Provisions Express 123 Shopping Street Cityville, State 54321 Country
    </address>
  </footer>
</div>
`;
